test/integration/pipeline/tx_pipeline_integration_test.cpp

```diff
@@ -219,3 +236,89 @@ TEST_F(TxPipelineIntegrationTest, TxPipelineTest) {
 
   validate();
 }
+
+/**
+ * @given Admin creates domains: [usabnk, ru, khm]
+ *        National Bank of America, creates domain usabnk (nba@usabnk) and asset usd,
+ * @when A user ivan in domain of Russia (ivan@ru) wants to possess and
+ *       transfer asset usd#usabnk to user tea in domain of Cambodia (tea@khm).
+ * @then Validate transactions processed.
+ */
+TEST_F(TxPipelineIntegrationTest, TransferAssetTestAmongDomainTest) {
+  const std::string DOMAIN_USABANK = "usabnk";
+  const std::string DOMAIN_RU = "ru";
+  const std::string DOMAIN_KHM = "khm";
+  const std::string USER_DEFAULT_ROLE = "user";
+  const std::string MONEY_CREATOR_ROLE = "money_creator";
+  const std::string ASSET_USD_NAME = "usd";
+  const std::string ASSET_USD_ID = ASSET_USD_NAME + "#" + DOMAIN_USABANK;
+  const std::string NBA_NAME = "nba";
+  const std::string IVAN_NAME = "ivan";
+  const std::string TEA_NAME = "tea";
+  const std::string ADMIN_ID = "admin@test";
+  const std::string NBA_ID = NBA_NAME + "@" + DOMAIN_USABANK;
+  const std::string IVAN_ID = IVAN_NAME + "@" + DOMAIN_RU;
+  const std::string TEA_ID = TEA_NAME + "@" + DOMAIN_KHM;
+
+  auto adminKeypair = iroha::KeysManagerImpl(ADMIN_ID).loadKeys().value();
+  auto nbaKeypair = createNewAccountKeypair(NBA_ID, "nba_pass").value();
+  auto ivanKeypair = createNewAccountKeypair(IVAN_ID, "ivan_pass").value();
+  auto teaKeypair = createNewAccountKeypair(TEA_ID, "tea_pass").value();
+
+  // Admin creates domains: [usabnk, khm, ru]
+  auto tx1 = TransactionGenerator().generateTransaction(
```

Comment:
You may move this "creation" stuff into additional fixture, imo the test is easier to read then